pkg/downloader/chart_downloader.go

```diff
@@ -139,15 +150,15 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 //		* If version is non-empty, this will return the URL for that version
 //		* If version is empty, this will return the URL for the latest version
 //		* If no version can be found, an error is returned
-func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, getter.Getter, error) {
+func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, getter.Getter, *repo.Entry, error) {
```

Comment:
If we're going to change this function, we should just refactor it to return the ChartRepository (`r`) instead of `r.Client` and `r.Entry` separately. It's backwards-incompatible, but third parties can still receive r.Client in their code. It would make this code a *lot* cleaner by not having to return `nil, nil` or `r.Client, r.Config` everywhere.

Reply:
Done.